inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/random_uniform/random_uniform_kernel_ref.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "random_uniform_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+namespace {
+
+using namespace kernel_selector;
+
+
+size_t GetFullShapeSize(const Tensor::DataTensor &tensor) {
+    return tensor.Batch().v * tensor.Feature().v * tensor.Z().v * tensor.Y().v * tensor.X().v * tensor.W().v;
+}
+
+int getStep(const random_uniform_params &params) {
+    return BytesPerElement(params.output.GetDType()) > 4 ? 2 : 4;
+}
+
+size_t GetGwsSize(const random_uniform_params &params) {
+    size_t shapeSize = GetFullShapeSize(params.output);
+    int step = getStep(params);
+    return (shapeSize >> (step >> 1)) + static_cast<int>(shapeSize % step > 0);
+}
+
+CommonDispatchData SetDefault(const random_uniform_params &params, const optional_params &) {
+    CommonDispatchData dispatchData;
+    dispatchData.gws = {GetGwsSize(params), 1, 1};
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    return dispatchData;
+}
+
+}  // namespace
+
+namespace kernel_selector {
+
+JitConstants RandomUniformKernelRef::GetJitConstants(const random_uniform_params &params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+    // When both seed values are equal to zero RandomUniform should generate non-deterministic sequence.
+    if (params.global_seed == 0 && params.op_seed == 0) {
+        std::srand(std::time(nullptr));
+        jit.AddConstant(MakeJitConstant("GLOBAL_SEED", std::rand()));
```

Comment:
`std::rand` is not guaranteed to be thread safe. though glibc one is.
modern portable solution would use https://en.cppreference.com/w/cpp/numeric/random/random_device#Example

Reply:
this rand is only about kernel initialization and rather corner case when both attributes are zero, do you really think we need thread safe guarantee here?

Reply:
as i mentioned, it is thread safe on our platform. i'd still use c++ facilities, they are recommended by standard instead of c ones https://eel.is/c++draft/c.math.rand#note-2